Tasks/SonarQubePostTest/PRCA/PostComments-Module.psm1

```diff
@@ -41,56 +72,143 @@ function InitPostCommentsModule
+        $script:project, # string project 
+        $script:pullRequest.Repository.Name, # string repositoryId 
+        $true, # bool? diffCommonCommit
+        $null, # int? top 
+        $null, # int? skip
+        $baseVersionDescriptor, 
+        $targetVersionDescriptor, 
+        $null, # object userState
+        [System.Threading.CancellationToken]::None # CancellationToken cancellationToken
+        ).Result
+    
+    if ($commitDiffs.ChangeCounts.Count -gt 0)
+    {
+        Write-Verbose "Found $($commitDiffs.ChangeCounts.Count) changed file(s) in the PR"
+        
+        $sourceFiles = $commitDiffs.Changes | 
+            Where-Object { ($_ -ne $null) -and ($_.Item.IsFolder -eq $false) }  | 
+            ForEach-Object { $_.Item.Path.Replace("\", "/") }
+    }
+
+    return $sourceFiles
+}
+
+#
+# Posts the discussion threads loaded with comments to the PR
+#
 function PostDiscussionThreads
 {
     param ([ValidateNotNull()][Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionThreadCollection]$threads)
     
     $vssJsonThreadCollection = New-Object -TypeName "Microsoft.VisualStudio.Services.WebApi.VssJsonCollectionWrapper[Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionThreadCollection]" -ArgumentList @(,$threads)
-    $script:discussionClient.CreateThreadsAsync($vssJsonThreadCollection, $null, [System.Threading.CancellationToken]::None).Result
+    [void]$script:discussionClient.CreateThreadsAsync($vssJsonThreadCollection, $null, [System.Threading.CancellationToken]::None).Result
     
     Write-Host "Posted $($threads.Count) discussion threads"
 }
 
 
-#endregion
-
-#region Private
+#
+# Retrieve existing discussion threads. 
+#
+function FetchDiscussionThreads
+{
+    $threadsDictionary = $script:discussionClient.GetThreadsAsync( @($script:artifactUri)).Result
+    
+    $threadList = New-Object "System.Collections.Generic.List[$script:discussionWebApiNS.DiscussionThread]"
+    
+    foreach ($threads in $threadsDictionary.Values)
+    {
+        if ($threads -ne $null)
+        {
+            Write-Verbose "Threads in current pair $($threads.Count)"
+            $threadList.AddRange($threads);
+        }
+    }
+    
+    Write-Verbose "Found $($threadList.Count) discussion thread(s)"
+    return $threadList;
+}
 
+#
+# Fetch existing discussion comments
+#
+function FetchDiscussionComments
+{
+    param ([System.Collections.Generic.List[Microsoft.VisualStudio.Services.CodeReview.Discussion.WebApi.DiscussionThread]]$discussionThreads)
+    
+    $comments = New-Object "System.Collections.Generic.List[$script:discussionWebApiNS.DiscussionComment]"
+    
+    foreach ($discussionThread in $discussionThreads)
+    {
+        $commentsFromThread = $script:discussionClient.GetCommentsAsync($discussionThread.DiscussionId).Result
+        if ($commentsFromThread -ne $null)
+        {
+            $comments.AddRange($commentsFromThread)
```

Comment:
The existing logic form the C# implementation gets the threads and the comments separately as lists, wheareas a dictionary seems more natural